packages/test/test-version-utils/src/versionUtils.ts

```diff
@@ -236,23 +236,25 @@ export const loadPackage = (modulePath: string, pkg: string) =>
  * generally negative to move back versions (eg. -1).
  * Note: If the requested number is a string then that will be the returned value
  */
-export function getRequestedRange(baseVersion: string, requested?: number | string): string {
+ export function getRequestedRange(baseVersion: string, requested?: number | string): string {
     if (requested === undefined || requested === 0) { return baseVersion; }
     if (typeof requested === "string") { return requested; }
     const version = new semver.SemVer(baseVersion);
-    // ask for prerelease in case we just bumped the version and haven't release the previous version yet.
-    if (version.major === 1) {
-        if (requested === -1) {
-            return "^0.59.0-0";
-        } else if (requested === -2) {
-            return "^0.58.0-0";
-        }
-    } else if (version.major === 2) {
-        if (requested === -1) {
-            return "^1.0.0-0";
-        } else if (requested === -2) {
-            return "^0.59.0-0";
-        }
+
+    // calculate requested major version number
+    const requestedMajorVersion = version.major + requested;
+    // if the major version number is bigger than 0 then return it as normal
+    if (requestedMajorVersion > 0) {
```

Comment:
@tylerbutler would this work for "internal" releases?

Reply:
for `2.0.0-internal.1.0.0` -- what is the expected previous major release? i'm assuming 1.0.0.

Reply:
Correct, the previous major of 2.0.0-internal.1.0.0 is 1.0

Reply:
What about 2.0.0-internal.2.0.0 ? Is it 2.0.0-internal.1.0.0?

Reply:
This is relevant to our e2e compat tests.

Reply:
previous major for `2.0.0-internal.2.0.0` is `2.0.0-internal.1.0.0`. I'll tweak the functionality and add few more test cases around it